core/utils_test.py

```diff
@@ -794,3 +794,85 @@ def test_url_open(self) -> None:
         self.assertEqual(response.getcode(), 200) # type: ignore[attr-defined]
         self.assertEqual(
             response.url, 'http://www.google.com') # type: ignore[attr-defined]
+
+    def test_get_random_int(self) -> None:
+        self.assertLess(utils.get_random_int(5), 5)
+        self.assertGreaterEqual(utils.get_random_int(5), 0)
+        self.assertLess(utils.get_random_int(True), 1)
+        self.assertGreaterEqual(utils.get_random_int(True), 0)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only positive integers allowed'):
+            utils.get_random_int(-1)
+
+    def test_get_random_choice(self) -> None:
+        list_instance = [1, 5, 9, 11, 15]
+        list_instance2: List[str] = []
+        self.assertIn(utils.get_random_choice(list_instance), list_instance)
+        with self.assertRaisesRegex( # type: ignore[no-untyped-call]
+            AssertionError, 'Only non-empty lists allowed'):
+            utils.get_random_choice(list_instance2)
+
+    def test_get_human_readable_time_string(self) -> None:
+        """Should we test for negative time."""
+        self.assertEqual('December 12 06:42:12', utils.get_human_readable_time_string(944980932342.38))  # pylint: disable=line-too-long
+
+    def test_generate_new_session_id(self) -> None:
+        self.assertEqual(24, len(utils.generate_new_session_id()))
+        self.assertEqual(type(utils.generate_new_session_id()).__name__, 'str')
```

Comment:
Are there any other requirements for a valid session ID? This test just checks that it's a 24-character string, which doesn't seem very strict

Reply:
There seems to be no other constraints to a valid session 
![SeshID](https://user-images.githubusercontent.com/81075125/159529201-b4ad8411-f4ad-4ac3-8fb0-cc2c7e402e7b.JPG)

We could maybe show that each string generated is random?

Reply:
Here's the code for `generate_random_string`:

https://github.com/oppia/oppia/blob/be72974f7d728b6084764ded43ff74abb836fb83/core/utils.py#L674-L683

So the string will always be a url-safe string of base64 characters

Reply:
I think we will have to change the test case for ```generate_random_string()``` as well because even that looks like this:
![random_string](https://user-images.githubusercontent.com/81075125/159862391-d58d6e9a-2d46-4b10-a227-3b22c72ad4fc.JPG)

Reply:
You're welcome to fix that test if you want! I won't require it before approving since that's not one of the tests you wrote for this PR, but fixing it certainly wouldn't hurt :)

Reply:
Done, I haven't made the changes in ```generate_random_string()``` as I'm not sure if the test case I've written is correct.

The logic behind my test case is that a byte string that is decoded in utf-8 when encoded again into utf-8 will use the same number of bits but when encoded with utf-16 it will use 16 bits which will lead to a different length. 

We can't however check if the string is decoded in utf-8 or ascii as os.urandom() gives a random byte which can be used both by ascii and utf-8